### PR TITLE
run-make-check.sh: let ctest generate XML output

### PR DESCRIPTION
**Pull request by tchaikov**
to enable XUnit plugin of jenkins to consume the ctest output and
publish it in the dashboard, we need to

* let ctest generate XML output instead of plain text output
* do not fail the test if any test case fails. this allows the publisher
  to do its job by checking the XML output.
* prevent ctest from compressing the output. see
  https://issues.jenkins.io/browse/JENKINS-21737

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: run-make-check.sh [hunk 1]
```diff
@@ -39,9 +43,16 @@ function run() {
     $DRY_RUN sudo /sbin/sysctl -q -w fs.aio-max-nr=$((65536 * $(nproc)))
 
     CHECK_MAKEOPTS=${CHECK_MAKEOPTS:-$DEFAULT_MAKEOPTS}
-    if ! $DRY_RUN ctest $CHECK_MAKEOPTS --output-on-failure; then
-        rm -fr ${TMPDIR:-/tmp}/ceph-asok.*
-        return 1
+    if in_jenkins; then
+        if ! ctest $CHECK_MAKEOPTS --no-compress-output --output-on-failure -T Test; then
+            # do not return failure, as the jenkins publisher will take care of this
+            rm -fr ${TMPDIR:-/tmp}/ceph-asok.*
```
Comment (epuertat) [inline run-make-check.sh:49]: Let's see how this works... Last time I tried, when no results were found the executor threw pretty ugly Java exceptions :/

**Review by epuertat — APPROVED** [review]
🎉

File: run-make-check.sh (hunk 1 above)
Comment (tchaikov) [reply run-make-check.sh:49]: @epuertat it works. see https://jenkins.ceph.com/job/ceph-pull-requests/72143/testReport/

Comment (epuertat) [reply run-make-check.sh:49]: Great. I really like the [OpenBlueOcean view]( https://jenkins.ceph.com/blue/organizations/jenkins/ceph-pull-requests/detail/ceph-pull-requests/72143/tests).

**Pull request merged**